man/rmdhunks/vcr-design.Rmd

```diff
@@ -0,0 +1,152 @@
+## Design
+
+This section explains `vcr`'s internal design and architecture.
+
+## Where vcr comes from and why R6
+
+`vcr` was "ported" from the Ruby gem (aka, library) of the same name[^1]. 
+Because it was ported from Ruby, an object-oriented programming language
+I thought it would be easier to use an object system in R that most
+closely resemble that used in Ruby (at least in my opinion). This 
+thinking lead to choosing [R6][]. The exported functions users interact
+with are not R6 classes, but are rather normal R functions. However,
+most of the internal code in the package uses R6. Thus, familiarity
+with R6 is important for people that may want to contribute to `vcr`,
+but not required at all for `vcr` users.
+
+## Principles
+
+### Mullet design: business up front, party in the back
+
+As described above, `vcr` uses R6 internally, but users interact with
+normal R functions. 
+
+```r
+functions == business
+R6 == party
+```
+
+The business and party could be reversed, depending on your perspective.
```

Comment:
I did not understand this image right away. Add a sentence about the hairstyle?

Reply:
AH, ok, not a universal thing, might change that